packages/grid/src/mouse-handlers/GridColumnSeparatorMouseHandler.ts

```diff
@@ -1,26 +1,58 @@
 /* eslint class-methods-use-this: "off" */
 import Grid from '../Grid';
 import GridMetricCalculator from '../GridMetricCalculator';
-import { ModelIndex, VisibleIndex } from '../GridMetrics';
+import type { ModelIndex, GridMetrics } from '../GridMetrics';
+import type GridModel from '../GridModel';
+import type { GridTheme } from '../GridTheme';
 import GridUtils, { GridPoint } from '../GridUtils';
-import GridSeparatorMouseHandler from './GridSeparatorMouseHandler';
+import GridSeparatorMouseHandler, {
+  GridSeparator,
+} from './GridSeparatorMouseHandler';
 
 class GridColumnSeparatorMouseHandler extends GridSeparatorMouseHandler {
-  static getColumnSeparatorIndex(
+  static getColumnSeparator(
     gridPoint: GridPoint,
-    grid: Grid,
-    checkAllowResize = true
-  ): VisibleIndex | null {
-    const theme = grid.getTheme();
-    if (checkAllowResize && !theme.allowColumnResize) {
+    metrics: GridMetrics,
+    model: GridModel,
+    theme: GridTheme
+  ): GridSeparator | null {
+    if (!theme.allowColumnResize) {
       return null;
     }
 
-    const { x, y } = gridPoint;
-    const { metrics } = grid;
-    if (!metrics) throw new Error('metrics not set');
+    const { x, y, columnHeaderDepth } = gridPoint;
 
-    return GridUtils.getColumnSeparatorIndex(x, y, metrics, theme);
+    const { modelColumns } = metrics;
+
+    const separatorIndex = GridUtils.getColumnSeparatorIndex(
+      x,
+      y,
+      metrics,
+      theme
+    );
+
+    if (
+      separatorIndex == null ||
+      columnHeaderDepth == null ||
+      columnHeaderDepth > 0
+    ) {
+      return null;
+    }
+
+    const columnIndex = modelColumns.get(separatorIndex);
+    const nextColumnIndex = modelColumns.get(separatorIndex + 1);
+    if (columnIndex == null || nextColumnIndex == null) {
+      return null;
+    }
+
+    if (
+      model.textForColumnHeader(columnIndex, columnHeaderDepth) !==
+      model.textForColumnHeader(nextColumnIndex, columnHeaderDepth)
+    ) {
+      return { index: separatorIndex, depth: columnHeaderDepth };
+    }
+
+    return null;
```

Comment:
This logic is incorrect... right now you cannot resize the last column. I think we said only allow resizing on the base column for now to make things simple? In which case it should be something like this:
```suggestion
    // TODO: Allow resizing of column groups as well. Right now just allow resizing from base columns
    if (
      separatorIndex == null ||
      columnHeaderDepth == null ||
      columnHeaderDepth > 0
    ) {
      return null;
    }

    const columnIndex = modelColumns.get(separatorIndex);
    if (columnIndex != null) {
      return { index: separatorIndex, depth: 0 };
    }

    return null;
```